File: leetcode/src/AddBinary.java
public class AddBinary {
    public static void main(String[] args) {
        AddBinary instance = new AddBinary();
        System.out.println(instance.addBinary("11", "1"));
        System.out.println(instance.addBinary("1010", "1011"));
    }
    public String addBinary(String a, String b) {
        StringBuffer result = new StringBuffer();
        int maxLength = Math.max(a.length(), b.length());
        int plus = 0;
        for (int i = 0; i < maxLength; i++) {
            char c1 = a.length() - 1 < i ? '0' : a.charAt(a.length() - 1 - i);
            char c2 = b.length() - 1 < i ? '0' : b.charAt(b.length() - 1 - i);
            int num = c1 - 48 + c2 - 48 + plus;
            if (num > 1) {
                plus = num / 2;
                num = num % 2;
            } else {
                plus = 0;
            }
            result.append(num);
        }
        if (plus > 0) {
            result.append(plus);
        }
        return result.reverse().toString();
    }
}

File: leetcode/src/python/add_binary.py
class AddBinary:
    def addBinary(self, a: str, b: str) -> str:
        result = ''
        max_len = max(len(a), len(b))
        left = 0
        for i in range(0, max_len):
            item_a = 0 if i > len(a) - 1 else int(a[len(a) - 1 - i])
            item_b = 0 if i > len(b) - 1 else int(b[len(b) - 1 - i])
            sum = item_a + item_b + left
            if sum > 1:
                result = str(sum - 2) + result
                left = 1
            else:
                result = str(sum) + result
                left = 0
        if left == 1:
            result = '1' + result
        return result

if __name__ == '__main__':
    instance = AddBinary()
    a = '1111'
    b = '1111'
    print(instance.addBinary(a, b))
